### ISSUE TITLE
layer-wise profiling of neural nets

### ISSUE THREAD

**Issue by pooyanjamshidi** [softsys4ai/neural-distiller#1]
We need to write code to profile and measuring energy and inference time layer-wise for deep nets. This provides us with a better understanding of energy consumption of neural nets in general and gives us a strategy to focus on layers that consume the most, so when we are pruning the teachers to build students, we would know where to start. There are magnitude-based and other criteria (Average Percentage of Zeros (APoZ), activation, etc) as well as Taylor expansion of lost function to estimate the change in the loss function, but we could also look at energy bottleneck to maybe come up with a better strategy.
